src/store/factory/client.js

```diff
@@ -94,14 +95,15 @@ function createEthereumClient (
   const ethClient = new Client()
   ethClient.addProvider(new EthereumRpcProvider({ uri: rpcApi }))
 
-  const legacyCoinType = 137
-  const { rskLegacyDerivation } = state
-  let coinType
-
+  const legacyCoinType = '137'
+  const ethereumNetworkCoinType = '60'
+  const { rskLegacyDerivation } = store.state
+  let coinType = ethereumNetwork.coinType
+  
   if (walletType === 'rsk_ledger') {
     coinType = legacyCoinType
-  } else {
-    coinType = rskLegacyDerivation ? legacyCoinType : ethereumNetwork.coinType
+  } else if (ethereumNetwork.name === 'rsk_mainnet') {
+    coinType = rskLegacyDerivation || state.rskLegacyDerivation ? legacyCoinType : ethereumNetworkCoinType
```

Comment:
this is getting confusing
rskLegacyDerivation === state.rskLegacyDerivation
why are there instances of both of them? And if they are the same value, what does `a || a ?` ... mean? When `a` is null, why do you need ternary to check it again?

Reply:
Yeah i notice that state and store.state are 2 different states and i just wanted to make sure i'm not gonna miss any value on both of them but yes i will remove state.rskLegacy if store.state is the real state